app/assets/javascripts/admin/controllers/admin-user-index.js.es6

```diff
@@ -54,6 +56,44 @@ export default Ember.Controller.extend(CanCheckEmails, {
     anonymize() { return this.get('model').anonymize(); },
     destroy() { return this.get('model').destroy(); },
 
+    toggleUsernameEdit() {
+      this.set('userUsernameValue', this.get('model.username'));
+      this.toggleProperty('editingUsername');
+    },
+
+    saveUsername() {
+      const self = this;
+
+      return ajax(`/users/${this.get('model.username').toLowerCase()}/preferences/username`, {
+        data: { new_username: this.get('userUsernameValue') },
+        type: 'PUT'
+      }).catch(function(e) {
+        bootbox.alert(I18n.t("generic_error_with_reason", {error: "http: " + e.status + " - " + e.body}));
+      }).finally(function() {
+        self.set('model.username', self.get('userUsernameValue'));
```

Comment:
this is not really correct, if it fails to save finally will still be called and username will be changed incorrectly 

instead the pattern you should follow. 

1. amend model username
2. issue the PUT request
3. if there is an error, revert model username
4. finally toggle the edit